core/src/main/java/org/keycloak/jose/jwk/JWKBuilder.java

```diff
@@ -62,75 +40,15 @@ public JWKBuilder algorithm(String algorithm) {
         return this;
     }
 
-    public JWK rs256(PublicKey key) {
-        algorithm(Algorithm.RS256);
-        return rsa(key);
-    }
-
-    public JWK rsa(Key key) {
-        return rsa(key, null, KeyUse.SIG);
-    }
-    
-    public JWK rsa(Key key, X509Certificate certificate) {
-        return rsa(key, Collections.singletonList(certificate), KeyUse.SIG);
-    }
-
-    public JWK rsa(Key key, List<X509Certificate> certificates) {
-        return rsa(key, certificates, null);
-    }
-
-    public JWK rsa(Key key, List<X509Certificate> certificates, KeyUse keyUse) {
-        RSAPublicKey rsaKey = (RSAPublicKey) key;
-
-        RSAPublicJWK k = new RSAPublicJWK();
-
-        String kid = this.kid != null ? this.kid : KeyUtils.createKeyId(key);
-        k.setKeyId(kid);
-        k.setKeyType(KeyType.RSA);
-        k.setAlgorithm(algorithm);
-        k.setPublicKeyUse(keyUse == null ? KeyUse.SIG.getSpecName() : keyUse.getSpecName());
-        k.setModulus(Base64Url.encode(toIntegerBytes(rsaKey.getModulus())));
-        k.setPublicExponent(Base64Url.encode(toIntegerBytes(rsaKey.getPublicExponent())));
-
-        if (certificates != null && !certificates.isEmpty()) {
-            String[] certificateChain = new String[certificates.size()];
-            for (int i = 0; i < certificates.size(); i++) {
-                certificateChain[i] = PemUtils.encodeCertificate(certificates.get(i));
-            }
-            k.setX509CertificateChain(certificateChain);
-        }
-
-        return k;
-    }
-
-    public JWK rsa(Key key, KeyUse keyUse) {
-        JWK k = rsa(key);
-        String keyUseString = keyUse == null ? DEFAULT_PUBLIC_KEY_USE.getSpecName() : keyUse.getSpecName();
-        if (KeyUse.ENC == keyUse) keyUseString = "enc";
-        k.setPublicKeyUse(keyUseString);
-        return k;
-    }
-
-    public JWK ec(Key key) {
-        return ec(key, DEFAULT_PUBLIC_KEY_USE);
+    @Override
+    public JWK okp(Key key) {
+        // not supported if jdk vesion < 17
+        return null;
```

Comment:
Better throw an exception, like `throw new UnsupportedOperationException("EdDSA algorithms not supported in this JDK version");`.

Reply:
Yes, I will fix it.